Xamarin.Forms.Platform.iOS/Platform.cs

```diff
@@ -144,6 +148,9 @@ Task INavigation.PushModalAsync(Page modal, bool animated)
 		{
 			EndEditing();
 
+			if (_appeared && IsModalPresentedOverContext(modal))
```

Comment:
I think we want to invert this check so that it's not super specific to OverContext. The pseudocode I would imagine for this would go like this

```C#

bool shouldFire = true;

if( iOS13OrHigher)
{
       if(presentationStyle == FullScreen) shouldFire = false; //this is mainly for backwards compatibility
}
else
{
      if(presentationStyle == Automatic) shouldFire = false;
      else if(presentationStyle == FullScreen) shouldFire = false; //this is mainly for backwards compatibility
}

if (_appeared && IsModalPresentedOverContext(modal))
```

This way it'll just work for whatever other presentation styles come up

Reply:
Then change `if (_appeared && IsModalPresentedOverContext(modal))` to `if (_appeared && shouldFire)`? And how about this one: https://github.com/xamarin/Xamarin.Forms/pull/6558/files/7bb10cdb8e6be4e42c9b12646188d6a4acec88dd#diff-7213603536449d008b596d1e2c2b15f4R116 ?